huaweicloud/services/identitycenter/resource_huaweicloud_identitycenter_account_assignment.go

```diff
@@ -0,0 +1,349 @@
+		deleteAccountAssignmentProduct = "identitycenter"
+	)
+	deleteAccountAssignmentClient, err := cfg.NewServiceClient(deleteAccountAssignmentProduct, region)
+	if err != nil {
+		return diag.Errorf("error creating Identity Center Client: %s", err)
+	}
+
+	instanceID := d.Get("instance_id").(string)
+	deleteAccountAssignmentPath := deleteAccountAssignmentClient.Endpoint + deleteAccountAssignmentHttpUrl
+	deleteAccountAssignmentPath = strings.ReplaceAll(deleteAccountAssignmentPath, "{instance_id}", instanceID)
+
+	deleteAccountAssignmentOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+	}
+	deleteAccountAssignmentOpt.JSONBody = utils.RemoveNil(buildAccountAssignmentBodyParams(d))
+	deleteAccountAssignmentResp, err := deleteAccountAssignmentClient.Request("POST",
+		deleteAccountAssignmentPath, &deleteAccountAssignmentOpt)
+	if err != nil {
+		return diag.Errorf("error deleting Identity Center account assignment: %s", err)
+	}
+
+	deleteAccountAssignmentRespBody, err := utils.FlattenResponse(deleteAccountAssignmentResp)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	requestID, err := jmespath.Search("account_assignment_deletion_status.request_id",
+		deleteAccountAssignmentRespBody)
+	if err != nil {
+		return diag.Errorf("error deleting Identity Center account assignment: request_id is not found in " +
+			"API response")
+	}
+
+	stateConf := &resource.StateChangeConf{
+		Pending: []string{"IN_PROGRESS"},
+		Target:  []string{"SUCCEEDED"},
+		Refresh: identityCenterStatusRefreshFunc(requestID.(string), instanceID, getRequestStatusHttpUrl,
+			"account_assignment_deletion_status.status", deleteAccountAssignmentClient),
+		Timeout:      d.Timeout(schema.TimeoutDelete),
+		Delay:        1 * time.Second,
+		PollInterval: 5 * time.Second,
+	}
+
+	_, err = stateConf.WaitForStateContext(ctx)
+	if err != nil {
+		return diag.Errorf("error waiting for request (%s) to complete: %s", requestID.(string), err)
+	}
+
+	return nil
+}
+
+func identityCenterStatusRefreshFunc(requestID, instanceID, getRequestStatusHttpUrl, searchExpression string,
+	client *golangsdk.ServiceClient) resource.StateRefreshFunc {
+	return func() (interface{}, string, error) {
+		getRequestStatusPath := client.Endpoint + getRequestStatusHttpUrl
+		getRequestStatusPath = strings.ReplaceAll(getRequestStatusPath, "{instance_id}", instanceID)
+		getRequestStatusPath = strings.ReplaceAll(getRequestStatusPath, "{request_id}", requestID)
+
+		getRequestStatusOpt := golangsdk.RequestOpts{
+			KeepResponseBody: true,
+		}
+		getRequestStatusResp, err := client.Request("GET", getRequestStatusPath, &getRequestStatusOpt)
+		if err != nil {
+			return nil, "", err
+		}
+		getRequestStatusRespBody, err := utils.FlattenResponse(getRequestStatusResp)
+		if err != nil {
+			return nil, "", err
+		}
+
+		status := utils.PathSearch(searchExpression, getRequestStatusRespBody, "")
+		return getRequestStatusRespBody, status.(string), nil
```

Comment:
suggest to make peding status a black list here, if it's not FAIL or SUCCEDED, it's IN PROGRESS in case of other pending status added.

Reply:
done